presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -346,9 +346,6 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     switch (insertExistingPartitionsBehavior) {
                         case APPEND:
                             checkState(!immutablePartitions);
-                            if (bucketNumber.isPresent()) {
```

Comment:
@dain what was rationale for this check?

Was this feature enabled by https://github.com/prestosql/presto/pull/822?

Reply:
At the time we believed that inserting into a bucketed unpartitoned table was illegal.  We believed that Hive required exactly one file per bucket, and since it is not possible to append to a file, it was illegal.  For all I know, it was illegal in old versions, but in the current versions it is supported.